### PR TITLE
fix(server): bound provider command output

### PR DESCRIPTION
**Pull request by Lucenx9**
## What changed

- Cap provider CLI stdout and stderr collection at 8 MiB each by default.
- Fail provider commands explicitly when either stream exceeds its limit.
- Preserve explicit smaller `maxBytes` overrides and avoid retrying deterministic overflow failures.

## Why

Provider probes shared a collector with no default output limit. A misbehaving CLI could therefore retain arbitrary stdout or stderr in the server process.

The provider collector now uses the same 8 MiB per-stream default as `ProcessRunner`. It drains both streams before reporting overflow so child processes can exit normally, including on Windows. Parsers never receive truncated JSON, version, or model output as a successful command result.

## Verification

- 62 focused provider tests pass.
- Server typecheck passes.
- Targeted lint and formatting checks pass.
- The regression test exceeds the default limit across three chunks, verifies all chunks were drained, and receives `ProviderCommandOutputLimitError`.
- The OpenCode process test writes 2 MiB to both stdout and stderr with a 64-byte limit and exits with an explicit overflow error.

## Checklist

- [x] This PR is focused on one reliability issue
- [x] I explained what changed and why
- [x] Before/after screenshots are not applicable because this PR has no UI changes
- [x] A video is not applicable because this PR has no animation or interaction changes

Model: gpt-5.6-sol
Harness: Codex in T3 Code

---

> [!NOTE]
> **Medium Risk**
> Touches shared provider probing and OpenCode inventory retry behavior; mis-sized limits could break legitimate large CLI output, but the change hardens memory use on a security-sensitive server path.
> 
> **Overview**
> Caps provider CLI **stdout/stderr** collection at **8 MiB** by default (with optional smaller `maxBytes`), replacing unbounded `collectStreamAsString` with `collectProviderCommandOutput` and a **`ProviderCommandOutputLimitError`** when either stream exceeds the budget. The collector still **drains** child output after the limit so processes can exit cleanly.
> 
> **Cursor** and **OpenCode** command paths now use the shared helper. Oversized `agent about` output surfaces as a health-check error instead of being parsed. OpenCode **fails** on limit breaches (no silent truncation), skips **retries** for output-limit failures during CLI inventory, while keeping model inventory when skill discovery alone overflows.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6d1fd2ce303220fc5ee6ec688ffa486a21b33fe4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Provider health checks now report a clear error when command output exceeds the supported 8 MiB limit.
  * Oversized command responses are handled consistently across providers, preventing incomplete output from being parsed as valid data.
  * Inventory checks no longer repeatedly retry commands that fail because their output exceeds the limit.
  * Command execution continues draining output before reporting size-limit failures, improving process cleanup and diagnostics.

### CODE REVIEW

**Comment by Lucenx9** [comment]
## Verification evidence

Tested commit [`6d1fd2ce`](https://github.com/pingdotgg/t3code/commit/6d1fd2ce3).

The review regression first went red because an oversized command still returned success:

```text
Test Files  1 failed (1)
Tests       1 failed | 4 passed (5)
```

On the published commit:

```sh
./node_modules/.bin/vp test run apps/server/src/provider/providerSnapshot.test.ts apps/server/src/provider/Layers/GrokProvider.test.ts apps/server/src/provider/Layers/CursorProvider.test.ts apps/server/src/provider/Drivers/GrokSkills.test.ts apps/server/src/provider/opencodeRuntime.inventory.test.ts
```

```text
Test Files  5 passed (5)
Tests       62 passed (62)
```

The focused tests prove these boundary conditions:

- A mock provider emits 8 MiB plus two bytes across three chunks. Collection reads all three chunks, then returns `ProviderCommandOutputLimitError`.
- A real child process writes 2 MiB to both stdout and stderr with a 64-byte limit. Both streams drain and the command returns an explicit overflow error.
- OpenCode records one oversized skill-discovery invocation, proving the deterministic overflow is not retried.
- Cursor rejects oversized `agent about` output instead of parsing the truncated response.

Server typecheck, targeted lint, formatting, and `git diff --check` all exit successfully. The [latest GitHub Actions run](https://github.com/pingdotgg/t3code/actions/runs/33699322992) verifies the published branch.

File: apps/server/src/provider/providerSnapshot.ts [hunk 1]
```diff
@@ -257,4 +258,7 @@ export const collectStreamAsString = <E>(
   stream: Stream.Stream<Uint8Array, E>,
   options?: { readonly maxBytes?: number | undefined },
 ): Effect.Effect<string, E> =>
-  collectUint8StreamText({ stream, ...options }).pipe(Effect.map((collected) => collected.text));
+  collectUint8StreamText({
+    stream,
+    maxBytes: options?.maxBytes ?? DEFAULT_PROVIDER_COMMAND_MAX_OUTPUT_BYTES,
+  }).pipe(Effect.map((collected) => collected.text));
```
[the remark this answers is not in the document]
Comment (Lucenx9) [reply apps/server/src/provider/providerSnapshot.ts]: Fixed in [`6d1fd2ce`](https://github.com/pingdotgg/t3code/commit/6d1fd2ce3). The shared command collector now drains both stdout and stderr, checks the truncation metadata, and returns `ProviderCommandOutputLimitError` before any parser receives partial output. Explicit smaller limits still apply, and OpenCode does not retry this deterministic failure.

The regression coverage includes the Grok/shared path, Cursor `agent about`, a real child process writing 2 MiB to both streams with a 64-byte cap, and an assertion that oversized OpenCode skill discovery runs only once. The focused suite passes 62/62.
